ipaserver/plugins/cert.py

```diff
@@ -858,11 +861,16 @@ def execute(self, csr, all=False, raw=False, chain=False, **kw):
                             "subject alt name type %s is forbidden "
                             "for non-user principals") % "RFC822Name"
                     )
+            elif isinstance(gn, cryptography.x509.general_name.IPAddress):
+                san_has_ipaddrs = True
```

Comment:
Like in ``DNSName`` case, you have prevent ``IPAddress`` SAN for user principals.

Reply:
Yes, good pickup @tiran.

Reply:
Yes, that was the intent.  I'm not familiar with this area at all, but wouldn't that be the right thing to do?  I.e., if a user can't "own" a DNS name, they can't own an IP address either.

If that's wrong, I'm going to have to rethink the logic for which IP addresses are considered acceptable for a particular certificate.

Reply:
I think that's right. I'd contain IPAddr entries to just hosts and services and if possible ensure that the IP address resolves to the hostname in question. This could be problematic for multi-homed hosts though. The managed_by attribute might be of use here assuming that all DNS hosts resolve into IPA hosts, in which case as long as a host/service requesting an IPAddr SAN is managed by the host that has the same name of the A/AAAA record then it is allowed.